pkg/iris/scan.go

```diff
@@ -158,9 +166,14 @@ func (s scanner) ScanNamespace(k8sApi *kubernetes.API, mapper Mapper, namespaces
 
 		// create kubernetes event for namespace
 		discoveryEvent := s.CreateDiscoveryEvent(namespace, mappedDeployments, &cluster, source, scope)
```

Comment:
what happens if any of the steps fail do you still need to call CreateEndReplay?

Reply:
If any of the steps fail, I guess this will not be called because it both discovery events and command events are sent in one request. I still need to understand properly how we can handle this

Reply:
so in theory we can revert it back to just /results once everything is finalised.

Reply:
I think you need something like try catch blocks as we have in java/kotlin

Reply:
@AlperCezibarakLeanix actually the error his handled in lines 122 ,155- 165

Reply:
golang doesn't have a try/catch

Reply:
than maybe you need to say that if err == nil {call endreplay event producer} something like that do you think it makes sense?